bin/perf_test.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import sys
+import time
+import random
+import string
+import socket
+from subprocess import call, check_output, STDOUT
+import xml.etree.ElementTree as ET
+
+# Carbon settings
+CARBON_SERVER = '0.0.0.0'
+CARBON_PORT = 2003
+
+sock = socket.socket()
+sock.connect((CARBON_SERVER, CARBON_PORT))
+
+BUCKET_NAMES = ['wednesday', 'gobig']
+METADATA_PROPS = ['make', 'color']
+
+def search(bucket_name, search_query):
+  cmd = "./search.sh %s %s" % (bucket_name, search_query)
+  print("  Exec cmd " + cmd)
+  return check_output(cmd, stderr=STDOUT, shell=True)
+
+def createFile(bucket_name, key):
+  print("Creating " + bucket_name + "/" + key)
+  call(["s3cmd", "put", "./a_file", "s3://%s/%s" % (bucket_name, key)])
+
+def send_metrics(metric_name, value):
+  message = "%s %d %d\n" % (metric_name, value, int(time.time()))
+  sock.sendall(message)
+
+def searchQueryForMetadata(metadata):
+  return ' AND '.join([ "userMd.\`x-amz-meta-%s\`=\\\\\\\"%s\\\\\\\"" % (k,v) for k,v in metadata.iteritems() ])
```

Comment:
the string escaping is so....any ideas for how we can lighten the burden on the user?

Reply:
It's HEAVY here, because we have to escape on Python, to be inserted inside another string that is passed to the command that passes to Livy, that will pass to Scala shell (as JSON, so has to be escaped one more time)